lib/src/native/realm_core.dart

```diff
@@ -1059,18 +1053,13 @@ class _RealmCore {
     });
   }
 
-  static void _logCallback(Pointer<Void> userdata, int levelAsInt, Pointer<Int8> message) {
+  static void _logCallback(Object userdata, int levelAsInt, Pointer<Int8> message) {
     final logger = Realm.logger;
+    final level = LevelExt.fromInt(levelAsInt);
 
-    try {
-      final level = LevelExt.fromInt(levelAsInt);
-
-      // Don't do expensive utf8 to utf16 conversion unless needed.
-      if (logger.isLoggable(level)) {
-        logger.log(level, message.cast<Utf8>().toDartString());
-      }
-    } finally {
-      _realmLib.realm_free(message.cast()); // always free the message.
+    // Don't do expensive utf8 to utf16 conversion unless needed.
+    if (logger.isLoggable(level)) {
+      logger.log(level, message.cast<Utf8>().toDartString());
     }
```

Comment:
I believe we should keep the `realm_free` call here

Reply:
or I am missing some new change elsewhere?

Reply:
The string memory is owned by the `realm_dart_sync_client_log_callback` trampoline, we don't need to free it in Dart.

Reply:
I guess this is owned now in the native struct copy